lib/matplotlib/dates.py

```diff
@@ -207,16 +207,19 @@ def _get_rc_timezone():
 
 def _to_ordinalf(dt):
     """
-    Convert :mod:`datetime` or :mod:`date` to the Gregorian date as UTC float
-    days, preserving hours, minutes, seconds and microseconds.  Return value
-    is a :func:`float`.
+    Convert :mod:`datetime`, :mod:`date` or :mod:`timedelta` to the Gregorian
+    date as UTC float days, preserving hours, minutes, seconds and
+    microseconds. Return value is a :func:`float`.
     """
     # Convert to UTC
     tzi = getattr(dt, 'tzinfo', None)
     if tzi is not None:
         dt = dt.astimezone(UTC)
         tzi = UTC
 
+    if isinstance(dt, datetime.timedelta):
```

Comment:
Another thing to note about this solution is that _ideally_ the way this would work is that `height` and `width` would apply to the values in the initial units / types, if possible.

Consider that if the plot's width is calculated by adding `width` to `dt_begin`, you automatically get the ability to add `dateutil.relativedelta.relativedelta` objects as well (for example, `relativedelta(weekday=MO(+3))` - the width being "from the initial value to three Mondays later). It's not high priority, but if you're solving this problem anyway, it's worth considering that leveraging the existing arithmetic framework for rich objects might be worth doing.